src/main/java/tc/oc/pgm/observers/tools/GamemodeTool.java

```diff
@@ -0,0 +1,90 @@
+package tc.oc.pgm.observers.tools;
+
+import com.google.common.collect.Lists;
+import java.util.List;
+import net.md_5.bungee.api.ChatColor;
+import net.md_5.bungee.api.chat.ClickEvent;
+import net.md_5.bungee.api.chat.HoverEvent;
+import org.apache.commons.lang.WordUtils;
+import org.bukkit.GameMode;
+import org.bukkit.Material;
+import tc.oc.component.Component;
+import tc.oc.component.render.ComponentRenderers;
+import tc.oc.component.types.PersonalizedText;
+import tc.oc.component.types.PersonalizedTranslatable;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.gui.InventoryGUI;
+import tc.oc.pgm.observers.ObserverTool;
+
+public class GamemodeTool implements ObserverTool {
+
+  @Override
+  public Component getName() {
+    return new PersonalizedTranslatable("observer.tools.gamemode");
+  }
+
+  @Override
+  public ChatColor getColor() {
+    return ChatColor.DARK_AQUA;
+  }
+
+  @Override
+  public List<String> getLore(MatchPlayer player) {
+    Component gamemode =
+        new PersonalizedText(WordUtils.capitalizeFully(player.getGameMode().name().toLowerCase()))
+            .color(ChatColor.AQUA);
+    Component lore =
+        new PersonalizedTranslatable("observer.tools.gamemode.lore", gamemode)
+            .getPersonalizedText()
+            .color(ChatColor.GRAY);
+    return Lists.newArrayList(ComponentRenderers.toLegacyText(lore, player.getBukkit()));
+  }
+
+  @Override
+  public Material getMaterial(MatchPlayer player) {
+    return isCreative(player) ? Material.SEA_LANTERN : Material.PRISMARINE;
+  }
+
+  @Override
+  public void onInventoryClick(InventoryGUI menu, MatchPlayer player) {
+    toggleObserverGameMode(player);
+    menu.refreshWindow(player);
+  }
+
+  public void toggleObserverGameMode(MatchPlayer player) {
+    player.setGameMode(getOppositeMode(player.getGameMode()));
+    if (player.getGameMode() == GameMode.SPECTATOR) {
+      player.sendWarning(getToggleMessage(), true);
+    } else if (isCreative(player)) {
+      player.getBukkit().performCommand("!");
```

Comment:
We should keep a comment here saying that this command is from WorldEdit and it unstucks the player.